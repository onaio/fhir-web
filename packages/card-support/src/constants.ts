// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
export const OPENSRP_URL_LOCATION_HIERARCHY = '/location/hierarchy';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
