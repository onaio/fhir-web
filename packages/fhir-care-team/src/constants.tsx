// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';
