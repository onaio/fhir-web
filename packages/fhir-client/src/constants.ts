export const dateFormat = 'yyyy-MM-dd';
