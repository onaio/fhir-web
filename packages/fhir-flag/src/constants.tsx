// urls
export const URL_ADMIN = '/admin';
export const URL_CLOSE_FLAGS = `${URL_ADMIN}/close-flags`;

// // magic strings

export const thatiMinutes = 30 * 60 * 1000;

// form fields
export const locationName = 'locationName' as const;
export const productName = 'productName' as const;
export const status = 'status' as const;
export const comments = 'comments' as const;
