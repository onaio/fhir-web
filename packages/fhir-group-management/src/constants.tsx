// url
export const ADD_EDIT_GROUP_URL = '/group/add-edit';
export const ADD_EDIT_COMMODITY_URL = '/commodity/add-edit';
export const LIST_GROUP_URL = '/groups/list';
export const LIST_COMMODITY_URL = '/commodity/list';

// magic strings
export const groupResourceType = 'Group';
export const listResourceType = 'List';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const type = 'type' as const;
export const unitOfMeasure = 'unitOfMeasure' as const;
