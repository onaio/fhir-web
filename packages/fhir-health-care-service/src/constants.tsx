// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTH_CARE_SERVICE_RESOURCE_TYPE = 'HealthcareService';
export const HEALTHCARES_ENDPOINT = 'HealthcareService/';
