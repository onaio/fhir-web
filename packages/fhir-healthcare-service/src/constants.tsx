// url
export const ADD_EDIT_HEALTHCARE_SERVICE_URL = '/healthcare/add-edit';
export const LIST_HEALTHCARE_URL = '/healthcare/list';

//
// constants
//
export const healthCareServiceResourceType = 'HealthcareService';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const comment = 'comment' as const;
export const extraDetails = 'extraDetails' as const;
export const providedBy = 'providedBy' as const;
