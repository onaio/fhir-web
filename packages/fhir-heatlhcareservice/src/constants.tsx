// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTHCARE_UUID = 'healthcare-uuid';
export const HEALTHCARES_GET = 'HealthcareService/';
export const HEALTHCARES_PUT = 'HealthcareService/';
export const HEALTHCARES_POST = 'HealthcareService/';

//
// Healthcares
//
export const ORGANIZATION_GET = 'Organization/';
