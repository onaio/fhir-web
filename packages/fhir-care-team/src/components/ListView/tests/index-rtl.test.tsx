import React from 'react';
import { Provider } from 'react-redux';
import { authenticateUser } from '@onaio/session-reducer';
import { CareTeamList } from '..';
import { Route, Router, Switch } from 'react-router';
import { createBrowserHistory } from 'history';
import nock from 'nock';
import * as reactQuery from 'react-query';
import { waitForElementToBeRemoved, fireEvent, render, cleanup } from '@testing-library/react';
import { store } from '@opensrp/store';
import { careTeam4214, careTeams } from './fixtures';
import { careTeamResourceType, URL_CARE_TEAM } from '../../../constants';
import { createMemoryHistory } from 'history';
import { RoleContext } from '@opensrp/rbac';
import { superUserRole } from '@opensrp/react-utils';

jest.mock('fhirclient', () => {
  return jest.requireActual('fhirclient/lib/entry/browser');
});

jest.mock('@opensrp/react-utils', () => {
  const actual = jest.requireActual('@opensrp/react-utils');

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const SearchForm = (props: any) => {
    const { onChangeHandler } = props;
    return (
      <div className="search-input-wrapper">
        <input onChange={onChangeHandler} data-testid="search-form"></input>
      </div>
    );
  };
  return {
    ...actual,
    SearchForm,
  };
});

const { QueryClient, QueryClientProvider } = reactQuery;

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});

jest.mock('@opensrp/notifications', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/notifications')),
}));

const history = createBrowserHistory();

const props = {
  fhirBaseURL: 'https://r4.smarthealthit.org/',
  history,
  careTeamPageSize: 5,
  location: {
    hash: '',
    pathname: `${URL_CARE_TEAM}`,
    search: '',
    state: {},
  },
  match: {
    isExact: true,
    params: { careTeamId: undefined },
    path: `${URL_CARE_TEAM}`,
    url: `${URL_CARE_TEAM}`,
  },
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AppWrapper = (props: any) => {
  return (
    <Provider store={store}>
      <QueryClientProvider client={queryClient}>
        <RoleContext.Provider value={superUserRole}>
          <Switch>
            <Route exact path={`${URL_CARE_TEAM}`}>
              {(routeProps) => <CareTeamList {...{ ...props, ...routeProps }} />}
            </Route>
            <Route exact path={`${URL_CARE_TEAM}/:careTeamId`}>
              {(routeProps) => <CareTeamList {...{ ...props, ...routeProps }} />}
            </Route>
          </Switch>
        </RoleContext.Provider>
      </QueryClientProvider>
    </Provider>
  );
};

describe('Care Teams list view', () => {
  beforeAll(() => {
    nock.disableNetConnect();
    store.dispatch(
      authenticateUser(
        true,
        {
          email: 'bob@example.com',
          name: 'Bobbie',
          username: 'RobertBaratheon',
        },
        // eslint-disable-next-line @typescript-eslint/naming-convention
        { api_token: 'hunter2', oAuth2Data: { access_token: 'sometoken', state: 'abcde' } }
      )
    );
  });
  afterAll(() => {
    nock.enableNetConnect();
  });

  afterEach(() => {
    jest.resetAllMocks();
    jest.clearAllMocks();
    jest.restoreAllMocks();
    cleanup();
  });

  it('renders correctly', async () => {
    nock(props.fhirBaseURL)
      .get(`/${careTeamResourceType}/_search`)
      .query({
        _total: 'accurate',
        _getpagesoffset: 0,
        _count: 20,
      })
      .reply(200, careTeams)
      .persist();

    const history = createMemoryHistory();
    history.push(URL_CARE_TEAM);

    const { getByText, queryByText } = render(
      <Router history={history}>
        <AppWrapper {...props} />
      </Router>
    );

    await waitForElementToBeRemoved(document.querySelector('.ant-spin'));

    expect(document.querySelector('title')).toMatchInlineSnapshot(`
      <title>
        FHIR Care Team
      </title>
    `);

    document.querySelectorAll('tr').forEach((tr, idx) => {
      tr.querySelectorAll('td').forEach((td) => {
        expect(td).toMatchSnapshot(`table row ${idx} page 1`);
      });
    });

    // view details
    nock(props.fhirBaseURL)
      .get(`/${careTeamResourceType}/_search`)
      .query({ _id: '308', _include: 'CareTeam:*' })
      .reply(200, careTeam4214)
      .persist();

    // target the initial row view details
    const dropdown = document.querySelector(
      'tbody tr:nth-child(1) [data-testid="action-dropdown"]'
    ) as Element;
    fireEvent.click(dropdown);

    const viewDetailsLink = getByText(/View Details/);
    expect(viewDetailsLink).toMatchInlineSnapshot(`
      <span>
        View Details
      </span>
    `);
    fireEvent.click(viewDetailsLink);
    expect(history.location.pathname).toEqual('/admin/CareTeams');
    expect(history.location.search).toEqual('?viewDetails=308');

    await waitForElementToBeRemoved(queryByText(/Fetching Care team/i));
    document.querySelectorAll('.display-block').forEach((block) => {
      expect(block).toMatchSnapshot('view details display block');
    });

    // close view details
    const closeButton = document.querySelector('[data-testid="cancel"]');
    fireEvent.click(closeButton);
  });
});
