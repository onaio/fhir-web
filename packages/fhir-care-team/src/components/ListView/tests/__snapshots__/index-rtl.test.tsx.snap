// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Care Teams list view renders correctly: Search 1 page 1 1`] = `
<td
  class="ant-table-cell"
>
  Care Team One
</td>
`;

exports[`Care Teams list view renders correctly: Search 1 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      href="/admin/CareTeam/edit/308"
    >
      <button
        class="ant-btn ant-btn-link m-0 p-1"
        type="button"
      >
        <span>
          Edit
        </span>
      </button>
    </a>
    <div
      class="ant-divider ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`Care Teams list view renders correctly: Search 2 page 1 1`] = `
<td
  class="ant-table-cell"
>
  Care Team Three
</td>
`;

exports[`Care Teams list view renders correctly: Search 2 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      href="/admin/CareTeam/edit/325"
    >
      <button
        class="ant-btn ant-btn-link m-0 p-1"
        type="button"
      >
        <span>
          Edit
        </span>
      </button>
    </a>
    <div
      class="ant-divider ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`Care Teams list view renders correctly: Search 3 page 1 1`] = `
<td
  class="ant-table-cell"
>
  Care Team Five
</td>
`;

exports[`Care Teams list view renders correctly: Search 3 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      href="/admin/CareTeam/edit/327"
    >
      <button
        class="ant-btn ant-btn-link m-0 p-1"
        type="button"
      >
        <span>
          Edit
        </span>
      </button>
    </a>
    <div
      class="ant-divider ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`Care Teams list view renders correctly: Search 4 page 1 1`] = `
<td
  class="ant-table-cell"
>
  test 33
</td>
`;

exports[`Care Teams list view renders correctly: Search 4 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      href="/admin/CareTeam/edit/330"
    >
      <button
        class="ant-btn ant-btn-link m-0 p-1"
        type="button"
      >
        <span>
          Edit
        </span>
      </button>
    </a>
    <div
      class="ant-divider ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`Care Teams list view renders correctly: Search 5 page 1 1`] = `
<td
  class="ant-table-cell"
>
  Care Team Six
</td>
`;

exports[`Care Teams list view renders correctly: Search 5 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      href="/admin/CareTeam/edit/328"
    >
      <button
        class="ant-btn ant-btn-link m-0 p-1"
        type="button"
      >
        <span>
          Edit
        </span>
      </button>
    </a>
    <div
      class="ant-divider ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`Care Teams list view renders correctly: Search 6 page 1 1`] = `
<td
  class="ant-table-cell"
>
  Care Team Four
</td>
`;

exports[`Care Teams list view renders correctly: Search 6 page 1 2`] = `
<td
  class="ant-table-cell"
>
  <span
    class="d-flex align-items-center"
  >
    <a
      href="/admin/CareTeam/edit/326"
    >
      <button
        class="ant-btn ant-btn-link m-0 p-1"
        type="button"
      >
        <span>
          Edit
        </span>
      </button>
    </a>
    <div
      class="ant-divider ant-divider-vertical"
      role="separator"
    />
    <span
      aria-label="more"
      class="anticon anticon-more ant-dropdown-trigger more-options"
      data-testid="action-dropdown"
      role="img"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        data-icon="more"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
        />
      </svg>
    </span>
  </span>
</td>
`;

exports[`Care Teams list view renders correctly: view details display block 1`] = `
<span
  class="ant-typography display-block"
>
  <strong>
    Name
  </strong>
</span>
`;

exports[`Care Teams list view renders correctly: view details display block 2`] = `
<span
  class="ant-typography ant-typography-secondary display-block"
>
  Peter Charlmers Care team
</span>
`;

exports[`Care Teams list view renders correctly: view details display block 3`] = `
<span
  class="ant-typography display-block"
>
  <strong>
    Identifier
  </strong>
</span>
`;

exports[`Care Teams list view renders correctly: view details display block 4`] = `
<span
  class="ant-typography ant-typography-secondary display-block"
/>
`;

exports[`Care Teams list view renders correctly: view details display block 5`] = `
<span
  class="ant-typography display-block"
>
  <strong>
    Status
  </strong>
</span>
`;

exports[`Care Teams list view renders correctly: view details display block 6`] = `
<span
  class="ant-typography ant-typography-secondary display-block"
>
  active
</span>
`;

exports[`Care Teams list view renders correctly: view details display block 7`] = `
<span
  class="ant-typography display-block"
>
  <strong>
    Participant
  </strong>
</span>
`;

exports[`Care Teams list view renders correctly: view details display block 8`] = `
<span
  class="ant-typography ant-typography-secondary display-block"
>
  
</span>
`;

exports[`Care Teams list view renders correctly: view details display block 9`] = `
<span
  class="ant-typography ant-typography-secondary display-block"
>
  
</span>
`;
